src/pal/src/locale/locale.cpp

```diff
@@ -359,7 +359,7 @@ static CFStringRef CFStringCreateMacFormattedLocaleName(LPCWSTR lpLocaleName)
         return NULL;
     }
 
-    CFStringAppendCharacters(cfMutableLocaleName, lpLocaleName, PAL_wcslen(lpLocaleName));
+    CFStringAppendCharacters(cfMutableLocaleName, (const UniChar*)lpLocaleName, PAL_wcslen(lpLocaleName));
```

Comment:
Could you please create UniChar * ToUniChar(WCHAR * ) helper method (or operator) and use it instead of the hard casts, so that we still get the types checked by the compiler?